rclcpp/include/rclcpp/node.hpp

```diff
@@ -392,6 +392,14 @@ class Node : public std::enable_shared_from_this<Node>
   std::vector<std::string>
   get_node_names() const;
 
+  RCLCPP_PUBLIC
+  std::vector<std::string>
+  get_node_namespaces() const;
```

Comment:
How do you know which namespace goes with which node? I think a `get_node_names_and_namespaces() -> std::vector<std::pair<std::string, std::string>>` would be a preferable addition.

Reply:
The get_node_namespaces function has been removed. get_node_names_and_namespaces() already has a signature of std::vector<std::pair<std::string, std::string>>, so not sure much needs to be changed here.